File: Lab/Lab5/Lab5.py
class Lab5(object):
    def isValidBST(self, root):
        return self.validBSTHelper(root, float('-inf'), float('inf'))

    def validBSTHelper(self, root, min, max):
        if root is None:
            return True
        if root.val <= min or root.val >= max:
            return False;
        return self.validBSTHelper(root.left, min, root.val) and self.validBSTHelper(root.right, root.val, max)

File: Labs/Lab 5/Lab5.java
public class Lab5 {
    public boolean isValidBST(TreeNode root) {
        return validBSTHelper(root, Integer.MIN_VALUE, Integer.MAX_VALUE);
    }
    public boolean validBSTHelper(TreeNode root, Integer min, Integer max){
        if (root == null){
            return true;
        }
        if (root.val <= min || root.val >= max){
            return false;
        }
        return validBSTHelper(root.left, min, root.val) && validBSTHelper(root.right, root.val, max);
    }
}
